pkg/infrastructure/services/custom_writer.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package services
```

Comment:
Same thing here. The client wrapper functions should be in the `client/prometheus` package instead. Actually, these functions does not make much sense to me, should be directly called in the prometheus package instead.

Reply:
I am moving these file under `client` because `CustomReader` & `CustomWriter` works for both Prometheus/Controller Client